hazelcast-jet-test-support/src/main/java/com/hazelcast/jet/testsupport/TestSupport.java

```diff
@@ -0,0 +1,278 @@
+     * with progress assertion enabled.
+     */
+    public static <T, U> void testProcessor(@Nonnull ProcessorSupplier supplier,
+                                            @Nonnull List<T> input, @Nonnull List<U> expectedOutput) {
+        testProcessor(processorFrom(supplier), input, expectedOutput, true);
+    }
+
+    /**
+     * Convenience for {@link #testProcessor(Processor, List, List, boolean)}
+     * with progress assertion enabled.
+     */
+    public static <T, U> void testProcessor(@Nonnull ProcessorMetaSupplier supplier,
+                                            @Nonnull List<T> input, @Nonnull List<U> expectedOutput) {
+        testProcessor(processorFrom(supplier), input, expectedOutput, true);
+    }
+
+    /**
+     * Convenience for {@link #testProcessor(Processor, List, List, boolean)}
+     * with progress assertion enabled.
+     */
+    public static <T, U> void testProcessor(@Nonnull Processor processor, @Nonnull List<T> input,
+                                            @Nonnull List<U> expectedOutput) {
+        testProcessor(processor, input, expectedOutput, true);
+    }
+
+    /**
+     * Convenience for {@link #testProcessor(Processor, List, List, boolean)}
+     * extracting the processor instance from the {@code supplier}.
+     */
+    public static <T, U> void testProcessor(
+            @Nonnull DistributedSupplier<Processor> supplier,
+            @Nonnull List<T> input,
+            @Nonnull List<U> expectedOutput,
+            boolean assertProgress
+    ) {
+        testProcessor(processorFrom(supplier), input, expectedOutput, assertProgress);
+    }
+
+    /**
+     * Convenience for {@link #testProcessor(Processor, List, List, boolean)}
+     * extracting the processor instance from the {@code supplier}.
+     */
+    public static <T, U> void testProcessor(
+            @Nonnull ProcessorSupplier supplier,
+            @Nonnull List<T> input,
+            @Nonnull List<U> expectedOutput,
+            boolean assertProgress
+    ) {
+        testProcessor(processorFrom(supplier), input, expectedOutput, assertProgress);
+    }
+
+    /**
+     * Convenience for {@link #testProcessor(Processor, List, List, boolean)}
+     * extracting the processor instance from the {@code supplier}.
+     */
+    public static <T, U> void testProcessor(
+            @Nonnull ProcessorMetaSupplier supplier,
+            @Nonnull List<T> input,
+            @Nonnull List<U> expectedOutput,
+            boolean assertProgress
+    ) {
+        testProcessor(processorFrom(supplier), input, expectedOutput, assertProgress);
+    }
+
+    /**
+     * An utility to test processors. It will initialize the processor
+     * instance, pass input items to it and assert the outbox contents.
+     * <p>
+     * For cooperative processors a 1-capacity outbox will be provided, which
+     * will additionally be full on every other processing method call. This
```

Comment:
It's an additional effect: I put `DUMMY_OBJ` to outbox to make it full, before every other process call.